tests/libvmi/vmi.go

```diff
@@ -107,22 +108,33 @@ func WithNodeSelectorFor(node *k8sv1.Node) Option {
 // WithUefi configures EFI bootloader and SecureBoot.
 func WithUefi(secureBoot bool) Option {
 	return func(vmi *v1.VirtualMachineInstance) {
-		vmi.Spec.Domain.Firmware = &v1.Firmware{
-			Bootloader: &v1.Bootloader{
-				EFI: &v1.EFI{
-					SecureBoot: pointer.BoolPtr(secureBoot),
-				},
-			},
+		if vmi.Spec.Domain.Firmware == nil {
+			vmi.Spec.Domain.Firmware = &v1.Firmware{}
 		}
+		if vmi.Spec.Domain.Firmware.Bootloader == nil {
+			vmi.Spec.Domain.Firmware.Bootloader = &v1.Bootloader{}
+		}
+		if vmi.Spec.Domain.Firmware.Bootloader.EFI == nil {
+			vmi.Spec.Domain.Firmware.Bootloader.EFI = &v1.EFI{}
+		}
```

Comment:
Nice :+1: !
This was actually a bug, since allocating resources resulting in overriding all `vmi.Spec.Domain.Firmware` specs makes absolutely no sense.